File: apps/courier/src/lib/format.ts
export const kr = (n: number) =>
  `${Math.round(Number(n) || 0).toLocaleString("sv-SE")} kr`;

export const km = (n: number) => `${(Number(n) || 0).toFixed(1).replace(".", ",")} km`;

/** Sekunder kvar tills en tidsstämpel (golvat vid 0). */
export const secondsLeft = (epochMs: number) => Math.max(0, Math.round((epochMs - Date.now()) / 1000));

export const dayLabel = (iso: string) => {
  const d = new Date(iso);
  const today = new Date();
  const yest = new Date();
  yest.setDate(today.getDate() - 1);
  const same = (a: Date, b: Date) => a.toDateString() === b.toDateString();
  if (same(d, today)) return "Idag";
  if (same(d, yest)) return "Igår";
  return d.toLocaleDateString("sv-SE", { weekday: "long", day: "numeric", month: "short" });
};

export const timeOfDay = (iso: string) =>
  new Date(iso).toLocaleTimeString("sv-SE", { hour: "2-digit", minute: "2-digit" });

File: mobile_apps/Delivera courier/lib/core/format.dart
import 'package:intl/intl.dart';

/// Formaterings-helpers — speglar `apps/courier/src/lib/format.ts` så att
/// kuriren ser exakt samma siffror i Flutter-appen som i webben.

final NumberFormat _kr = NumberFormat.decimalPattern('sv_SE');

/// "120 kr" — heltal, svensk tusentalsavgränsare.
String kr(num n) => '${_kr.format((n).round())} kr';

/// "2,4 km" — en decimal, svenskt komma.
String km(num n) => '${n.toStringAsFixed(1).replaceAll('.', ',')} km';

/// Sekunder kvar tills en epoch-ms-tidsstämpel (golvat vid 0).
int secondsLeft(int epochMs) {
  final diff = (epochMs - DateTime.now().millisecondsSinceEpoch) / 1000;
  return diff < 0 ? 0 : diff.round();
}

/// "Idag" / "Igår" / "måndag 3 jun".
String dayLabel(DateTime d) {
  final now = DateTime.now();
  final today = DateTime(now.year, now.month, now.day);
  final that = DateTime(d.year, d.month, d.day);
  final diff = today.difference(that).inDays;
  if (diff == 0) return 'Idag';
  if (diff == 1) return 'Igår';
  return DateFormat('EEEE d MMM', 'sv_SE').format(d);
}

/// "18:09" — tid på dygnet.
String timeOfDay(DateTime d) => DateFormat('HH:mm').format(d);

/// "8 jun" — kort datum (för datumintervall).
String dateShort(DateTime d) => DateFormat('d MMM', 'sv_SE').format(d);

/// "8 jun – 14 jun" eller "14 jun" om start = slut.
String dateRangeLabel(DateTime start, DateTime end) {
  final a = DateTime(start.year, start.month, start.day);
  final b = DateTime(end.year, end.month, end.day);
  if (a == b) return dateShort(a);
  return '${dateShort(a)} – ${dateShort(b)}';
}

/// "11 min" — minuter.
String minutes(num n) => '${n.round()} min';
